### PR TITLE
LPS-197691 Fixed creation of properties that come from model definition of a related object

### PR DESCRIPTION
**Pull request by jalber786**
**What's changed?:**
- Added new code that, in case the field of the related object searched is not found in the object properties, acceses the Object Definition model fields.

**What's new?**
- Added new tests that checks if the behavior is fixed.

See the following **Jira Ticket**: [LPS-197691](https://liferay.atlassian.net/browse/LPS-197691)

### CODE REVIEW

**Comment by jalber786** [comment]
ci:test:relevant

File: modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java [hunk 1]
```diff
@@ -138,14 +141,25 @@ private Map<String, Object> _getObjectEntryProperties(
...
+			ObjectEntry objectEntry, APIApplication.Property property,
+			List<String> relationshipsNames)
+		throws Exception {
 
 		if (relationshipsNames.isEmpty()) {
 			Map<String, Object> objectEntryProperties =
 				objectEntry.getProperties();
 
-			return objectEntryProperties.get(property.getSourceFieldName());
+			Object object = objectEntryProperties.get(
```
Comment (ccorreagg) [inline modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/EndpointHelper.java:152]: I would purpose to implement in REST Builder a method in the DTO (in this case, in ObjectEntry) to get a property based on its name. As a property is set with `x-json-map: true`, that means Jackson will include any unknown property into the map, so we have the power in REST Builder to know if a property is given by its field or by the map.

Something like:

```
public Object getProperty(String propertyName) {
	if (Objects.equals(propertyName, "actions")) {
		return actions;
	} else if (Objects.equals(propertyName, "auditEvents")) {
		return auditEvents;
	} else if (Objects.equals(propertyName, "creator")) {
		return creator;
		
	[...]
		
	} else if (Objects.equals(propertyName, "externalReferenceCode")) {
		return externalReferenceCode;
	} else {
		return properties.get(propertyName);
	}
}
```


What do you think @4lejandrito ??

**Comment by jalber786** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
Maybe I am wrong, but do we agree to modify this PR in order to perform the changes in the REST builder? @jalber786 @4lejandrito

**Comment by magjed4289** [comment]
Test failures are not related. OK to manually forward. ✅

**Comment by jalber786** [comment]
ci:forward:force

> Maybe I am wrong, but did we agree to modify this PR in order to perform the changes in the REST builder? @jalber786 @4lejandrito

There's already a tech debt task for it. I'm working on it right now but we considered this to be a big enough bug to fix it and then refactor it

**Pull request closed without merging**